libyul/Object.cpp

```diff
@@ -129,33 +141,64 @@ std::set<YulString> Object::qualifiedDataNames() const
 	return qualifiedNames;
 }
 
-std::vector<size_t> Object::pathToSubObject(YulString _qualifiedName) const
+void Object::visitPath(YulString _qualifiedName, std::function<bool(Object const*)> const& _visitor) const
 {
-	yulAssert(_qualifiedName != name, "");
-	yulAssert(subIndexByName.count(name) == 0, "");
-
-	if (boost::algorithm::starts_with(_qualifiedName.str(), name.str() + "."))
-		_qualifiedName = YulString{_qualifiedName.str().substr(name.str().length() + 1)};
 	yulAssert(!_qualifiedName.empty(), "");
 
 	std::vector<std::string> subObjectPathComponents;
 	boost::algorithm::split(subObjectPathComponents, _qualifiedName.str(), boost::is_any_of("."));
 
-	std::vector<size_t> path;
 	Object const* object = this;
-	for (std::string const& currentSubObjectName: subObjectPathComponents)
+
+	// Iterate over sub objects ignoring the top-level object in the path.
+	for (std::string const& currentSubObjectName: subObjectPathComponents | ranges::views::drop(1))
```

Comment:
I'm not completely sure if it is fine to drop the first path element in the context of `pathToSubObject`.

Reply:
Actually it is not, because of the use of `starts_with` and `substr` there. I will fix it.

Reply:
It should be fixed now:)